ktor-server/ktor-server-core/src/io/ktor/content/StaticContentResolution.kt

```diff
@@ -27,7 +27,7 @@ fun ApplicationCall.resolveResource(path: String,
         when (url.protocol) {
             "file" -> {
                 val file = File(decodeURLPart(url.path))
-                return if (file.exists()) LocalFileContent(file, mimeResolve(file.extension)) else null
+                return if (file.exists() && file.isFile) LocalFileContent(file, mimeResolve(file.extension)) else null
```

Comment:
we can replace exists check with `isFile` check as it will return false for non-existing path as well. It will avoid double native function invocation inside of `java.io.File`

Reply:
Good catch, I'll remove the `.exists()` call.